promql/engine.go

```diff
@@ -705,16 +734,21 @@ func (ev *evaluator) rangeEval(f func([]Value, *EvalNodeHelper) Vector, exprs ..
 	enh := &EvalNodeHelper{out: make(Vector, 0, biggestLen)}
 	seriess := make(map[uint64]Series, biggestLen) // Output series by series hash.
 	for ts := ev.startTimestamp; ts <= ev.endTimestamp; ts += ev.interval {
+		// Reset num. of samples in memory after each timestamp.
+		tempNumSamples = ev.currentSamples
```

Comment:
This can go above the for, and then you can reset it here - which is more natural than doing it at the end of the loop